soh/soh/resource/importer/scenecommand/SetMeshFactory.cpp

```diff
@@ -36,31 +36,39 @@ SetMeshFactory::ReadResource(std::shared_ptr<Ship::ResourceInitData> initData, s
     for (int32_t i = 0; i < polyNum; i++) {
         if (setMesh->meshHeader.base.type == 0) {
             PolygonDlist dlist;
+            dlist.opa = nullptr;
+            dlist.xlu = nullptr;
 
             int32_t polyType = reader->ReadInt8(); // Unused
             std::string meshOpa = reader->ReadString();
             std::string meshXlu = reader->ReadString();
 
-            auto opaRes = Ship::Context::GetInstance()->GetResourceManager()->LoadResourceProcess(meshOpa.c_str());
-            auto xluRes = Ship::Context::GetInstance()->GetResourceManager()->LoadResourceProcess(meshXlu.c_str());
-            dlist.opa = meshOpa != "" ? (Gfx*)(opaRes ? opaRes->GetRawPointer() : nullptr) : 0;
-            dlist.xlu = meshXlu != "" ? (Gfx*)(xluRes ? xluRes->GetRawPointer() : nullptr) : 0;
+            // Enables alt-toggling support by setting maintained c_str references to DList resource after pushing to vector the first
+            // Defers resource loading later in game when the scene is drawn
+            if (meshOpa != "") {
+                meshOpa = "__OTR__" + meshOpa;
+                setMesh->opaPaths.push_back(meshOpa);
+                dlist.opa = (Gfx*)setMesh->opaPaths.back().c_str();
+            }
+            if (meshXlu != "") {
+                meshXlu = "__OTR__" + meshXlu;
+                setMesh->xluPaths.push_back(meshXlu);
+                dlist.xlu = (Gfx*)setMesh->xluPaths.back().c_str();
+            }
 
-            setMesh->opaPaths.push_back(meshOpa);
-            setMesh->xluPaths.push_back(meshXlu);
             setMesh->dlists.push_back(dlist);
         } else if (setMesh->meshHeader.base.type == 1) {
             PolygonDlist pType;
+            pType.opa = nullptr;
+            pType.xlu = nullptr;
 
             setMesh->meshHeader.polygon1.format = reader->ReadUByte();
+
+            // These strings are the same that are read and used below. Not sure why they get exported twice from the exporter.
+            // We ignore these in favor of using the ones below.
```

Comment:
if we're not using them and just need to advance the reader we can just do
```cpp
reader->ReadString();
reader->ReadString();
```
and not actually read into anything